### PR TITLE
Feat(auth): Handle resetPassword and Invite user

### PR DESCRIPTION
**Pull request by arimet**
## Problem

As a new user, I want to know when I'm invited to join the CRM and configure my own password

## Solution

Implement setPassword and Invite user via Supabase. Handle in HashRouter

### Reset
1. Launch demo
2. From Login page go to ResetPasswordPage. 
3. Send recovery email and go to http://localhost:54324/
4. Click on link and reset password

### Invite user
1. Go on Sales Create
2. Create a new sales and go to http://localhost:54324/
3. Click on mail and set password

### CODE REVIEW

**Review by jonathan-marmelab — APPROVED** [review]

File: supabase/functions/users/index.ts [hunk 1]
```diff
@@ -35,10 +35,12 @@ async function inviteUser(req: Request) {
         await supabaseAdmin.auth.admin.createUser({
             email,
             password,
-            email_confirm: true,
             user_metadata: { first_name, last_name },
         });
 
+    const { data: emailData, error: emailError } =
```
Comment (jonathan-marmelab) [inline supabase/functions/users/index.ts:41]: lint error

File: package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
         "papaparse": "^5.4.1",
         "prop-types": "^15.8.1",
         "ra-data-fakerest": "^5.1.1",
-        "ra-supabase": "^3.0.0",
+        "ra-supabase": "^3.0.1",
```
Comment (slax57) [inline package.json:22]: nitpick: there is no 3.0.1. You should require a 3.1.0 instead.


```suggestion
        "ra-supabase": "^3.1.0",
```

File: supabase/functions/users/index.ts [hunk 3]
```diff
@@ -35,11 +35,13 @@ async function inviteUser(req: Request) {
         await supabaseAdmin.auth.admin.createUser({
             email,
             password,
-            email_confirm: true,
             user_metadata: { first_name, last_name },
         });
 
-    if (!data?.user || userError) {
+    const { _, error: emailError } =
```
Comment (jonathan-marmelab) [inline supabase/functions/users/index.ts:41]: **nit-pick:** The underscore is probably not necessary

```suggestion
    const { error: emailError } =
```

File: supabase/config.toml [hunk 4]
```diff
@@ -71,7 +71,7 @@ enabled = true
 # in emails.
 site_url = "http://localhost:5173/"
 # A list of *exact* URLs that auth providers are permitted to redirect to post authentication.
-additional_redirect_urls = ["https://localhost:5173/auth-callback/index.html"]
+additional_redirect_urls = ["https://localhost:5173/auth-callback.html"]
```
Comment (slax57) [inline supabase/config.toml:74]: don't you need to update the documentation following this change?

File: supabase/templates/invite.html [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    <body>
+        <div class="container">
+            <div class="header">
+                <h2>Confirm your account</h2>
+            </div>
+            <div class="content">
+                <p>Welcome to Atomic CRM.</p>
+                <p>
+                    You've been invited to join the (Name) organization. Confirm
+                    your address below:
```
Comment (slax57) [inline supabase/templates/invite.html:64]: This too looks like something that we should document (replacing the organization name in the template)

File: src/sales/SalesCreate.tsx [hunk 6]
```diff
@@ -39,13 +37,7 @@ export function SalesCreate() {
             <Card>
                 <SimpleForm onSubmit={onSubmit as SubmitHandler<any>}>
                     <Typography variant="h6">Create sale person</Typography>
-                    <SalesInputs>
-                        <PasswordInput
-                            source="password"
-                            validate={required()}
-                            helperText={false}
-                        />
```
Comment (slax57) [inline src/sales/SalesCreate.tsx:47]: Doesn't this break the fakerest implementation? :thinking:

**Comment by slax57** [comment]
Also, I think I found a small bug.
When you invite a new Admin, after they fill in their new password, they don't see the 'Users' view right away.
They need to sign out / sign in again to see it.

**Review by jonathan-marmelab — APPROVED** [review]

**Pull request merged**